src/encode_task_bwa.py

```diff
@@ -213,30 +177,12 @@ def main():
         bam = bwa_se(args.fastqs[0], 
             bwa_index_prefix, args.nth, args.out_dir)
 
-    # initialize multithreading
-    log.info('Initializing multi-threading...')
-    num_process = min(2,args.nth)
-    log.info('Number of threads={}.'.format(num_process))
-    pool = multiprocessing.Pool(num_process)
-    
-    log.info('Running samtools index...')
-    ret_val1 = pool.apply_async(
-        samtools_index, (bam, args.out_dir))
-
-    log.info('Running samtools flagstat...')
-    ret_val2 = pool.apply_async(
-        samtools_flagstat, (bam, args.out_dir))
-
-    bai = ret_val1.get(BIG_INT)
-    flagstat_qc = ret_val2.get(BIG_INT)
-
-    log.info('Closing multi-threading...')
-    pool.close()
-    pool.join()
-
     log.info('Removing temporary files...')
     rm_f(temp_files)
 
+    log.info('Checking if BAM file is empty...')
+    assert(int(run_shell_cmd('samtools view -c {}'.format(bam))))
```

Comment:
asserts should be used for tests and debugging, not as an error mechanism. It's more appropriate to do the following:
```suggestion
    if not int(run_shell_cmd('samtools view -c {}'.format(bam))):
        raise ValueError('BAM file is empty, no reads found.')
```

Reply:
thanks, fixed.